src/db.rs

```diff
@@ -141,11 +146,29 @@ pub async fn list_all_hosts(db: &Client) -> Result<Vec<HostSummary>> {
     cursor_extended.try_collect().await.map_err(Debug)
 }
 
-pub async fn list_registered_hosts(db: &Client) -> Result<Vec<bson::Bson>> {
+// This gets a list of all HPs including those not SSH'd
+pub async fn list_registered_hosts(db: &Client, cutoff: u64) -> Result<Vec<bson::Bson>> {
     // Retrieve all holoport statuses and format for an API response
     let hp_status: Collection<Host> = db
         .database("host_statistics")
         .collection("holoports_status");
 
-    hp_status.distinct("name", None, None).await.map_err(Debug)
+    let ms = get_cutoff_timestamp(cutoff);
+
+    let filter = doc!{"timestamp": {"$gte": ms.to_string()}};  
+
+    hp_status.distinct("name", filter, None).await.map_err(Debug)
 }
+
+// Helper function to get cutoff timestamp for filter
+fn get_cutoff_timestamp(days: u64) -> i64 {
+    let duration = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .expect("SystemTime should be after unix epoch");
+    let one_week = Duration::from_secs(60 * 60 * 24 * days);
+    let one_week_ago = duration
+        .checked_sub(one_week)
+        .expect("SystemTime should be at least one week after unix epoch");
+    use std::convert::TryInto;
```

Comment:
I do not really understand the logic of checking one week here. Scenario here is as follows:
- user is calling API`/list-available` or `/registered`
- they will pass a query parameter `days`, which means per documentation - give me entries not older than <days>
- In this function there's a query parameter value validation happening
What I am saying is are there any limits on that parameter? What if those limits are not met? Are we panicking? This will return 500 to the user, which is semantically wrong.

Reply:
It's not checking one_week here btw.

`checked_sub` just means subtract but check that it doesn't become a negative timestamp.